File: oj-web/src/enums/oj/topic/TopicJudgeType.ts
import type { BaseEnum } from '@/enums/base';

// 主题判题类型枚举接口
interface TopicJudgeTypeEnum extends BaseEnum {
  tooltip?: string;
  text?: string;
}

/**
 * 主题判题类型枚举
 * 用于标识不同的判题类型
 */
export const TopicJudgeType: { [key: string]: TopicJudgeTypeEnum } = {
  /**
   * ACM 判题模式
   */
  ACM: {
    code: 0,
    name: 'acm',
    tooltip: 'ACM 判题模式',
    text: 'ACM 判题模式'
  },

  /**
   * OI 判题模式
   */
  OI: {
    code: 1,
    name: 'oi',
    tooltip: 'OI 判题模式',
    text: 'OI 判题模式'
  }
};

/**
 * 根据code获取枚举
 * @param code
 * @returns 枚举值或null
 */
export function matchCode(code: number): TopicJudgeTypeEnum | null {
  return Object.values(TopicJudgeType).find(item => item.code === code) || null;
} 

File: glowxq-api/business/business-oj/src/main/java/com/glowxq/oj/topic/enums/TopicJudgeType.java
package com.glowxq.oj.topic.enums;

import com.glowxq.core.common.enums.base.BaseType;
import com.glowxq.core.common.exception.common.AlertsException;
import lombok.AllArgsConstructor;
import lombok.Getter;

/**
 * @author glowxq
 * @version 1.0
 * @date 2025/3/18
 */
@Getter
@AllArgsConstructor
public enum TopicJudgeType implements BaseType {
    ACM(0, "acm"),

    OI(1, "oi");

    private final Integer type;

    private final String name;

    /**
     * 根据code获取枚举
     *
     * @return 对应的枚举实例，未找到返回null
     */
    public static TopicJudgeType matchCode(Integer code) {
        for (TopicJudgeType type : TopicJudgeType.values()) {
            if (type.getCode().equals(code)) {
                return type;
            }
        }
        throw new AlertsException("编程题类型转换错误:" + code);
    }

    @Override
    public Object getCode() {
        return type;
    }
}
